src/Catalyst.KBucket.Tests/SplitTest.cs

```diff
@@ -0,0 +1,135 @@
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Catalyst.KBucket
+{
+    /// <summary>
+    ///   From https://github.com/tristanls/k-bucket/blob/master/test/split.js
+    /// </summary>
+    [TestClass]
+    public class SplitTest
+    {
+        [TestMethod]
+        public void OneContactDoesNotSplit()
+        {
+            var kBucket = new KBucket<Contact>
+            {
+                new Contact("a")
+            };
+            Assert.IsNull(kBucket.Root.Left);
+            Assert.IsNull(kBucket.Root.Right);
+            Assert.IsNotNull(kBucket.Root.Contacts);
+        }
+
+        [TestMethod]
+        public void MaxContactsPerNodeDoesNotSplit()
+        {
+            var kBucket = new KBucket<Contact>();
+            for (var i = 0; i < kBucket.ContactsPerBucket; ++i) kBucket.Add(new Contact(i));
+
+            Assert.IsNull(kBucket.Root.Left);
+            Assert.IsNull(kBucket.Root.Right);
+            Assert.IsNotNull(kBucket.Root.Contacts);
+        }
+
+        [TestMethod]
+        public void MaxContactsPerNodePlusOneDoetSplit()
+        {
+            var kBucket = new KBucket<Contact>();
+            for (var i = 0; i < kBucket.ContactsPerBucket + 1; ++i) kBucket.Add(new Contact(i));
+
+            Assert.IsNotNull(kBucket.Root.Left);
+            Assert.IsNotNull(kBucket.Root.Right);
+            Assert.IsNull(kBucket.Root.Contacts);
+        }
+
+        [TestMethod]
+        public void SplitNodesContainsAllContacts()
+        {
+            var kBucket = new KBucket<Contact>
+            {
+                LocalContactId = new byte[] {0x00}
+            };
+            var contacts = new List<Contact>();
+            for (var i = 0; i < kBucket.ContactsPerBucket + 1; ++i)
+            {
+                var contact = new Contact((byte) i);
+                contacts.Add(contact);
+                kBucket.Add(contact);
+            }
+
+            foreach (var contact in contacts) Assert.IsTrue(kBucket.Contains(contact));
+        }
+
+        [TestMethod]
+        public void FarAway()
+        {
+            var kBucket = new KBucket<Contact>
+            {
+                LocalContactId = new byte[] {0x00}
+            };
+            for (var i = 0; i < kBucket.ContactsPerBucket + 1; ++i) kBucket.Add(new Contact((byte) i));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'for' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)